File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/order/OrderGroupChoiceResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.order;

import com.ovvium.services.model.bill.IssueStatus;
import com.ovvium.services.model.bill.OrderGroupChoice;
import com.ovvium.services.model.bill.ServiceTime;
import com.ovvium.services.web.controller.bff.v1.transfer.response.common.ResourceIdResponse;
import com.ovvium.services.web.controller.bff.v1.transfer.response.product.ProductResponse;
import lombok.Getter;

@Getter
public final class OrderGroupChoiceResponse extends ResourceIdResponse {

	private final ProductResponse product;
	private final IssueStatus issueStatus;
	private final ServiceTime serviceTime;
	private final long orderTime;
	private final String notes;

	public OrderGroupChoiceResponse(OrderGroupChoice choice, ProductResponse productResponse) {
		super(choice);
		this.issueStatus = choice.getIssueStatus();
		this.product = productResponse;
		this.serviceTime = choice.getServiceTime();
		this.orderTime = choice.getCreated().toEpochMilli();
		this.notes = choice.getNotes().orElse(null);
	}

}

File: ovvium-pos/src/app/model/response/OrderGroupChoiceResponse.ts
import { ResourceIdResponse } from "app/model/response/ResourceIdResponse";
import { ProductResponse } from "app/model/response/ProductResponse";

export class OrderGroupChoiceResponse extends ResourceIdResponse {

    product: ProductResponse;
    issueStatus: string;
    serviceTime: string;
    orderTime: number;
    notes: string;

    constructor(response: OrderGroupChoiceResponse) {
        super(response.id);
        this.product = response.product;
        this.issueStatus = response.issueStatus;
        this.serviceTime = response.serviceTime;
        this.orderTime = response.orderTime;
        this.notes = response.notes;
    }    

}
